### PR TITLE
add JAVA_HOME env variable check

### PR DESCRIPTION
**Pull request by hyeung-trulia**
add JAVA_HOME env variable check before running path.resolve with process.env.JAVA_HOME

### CODE REVIEW

**Comment by jlipps** [comment]
Unfortunately because of the way JS's async works, if you want to do this check you'll need to put everything after the check inside the check's callback. Does that make sense?

File: lib/adb.js [hunk 1]
```diff
@@ -494,16 +494,21 @@ ADB.prototype.checkApkCert = function (apk, pkg, cb) {
 };
 
 ADB.prototype.checkCustomApkCert = function (apk, pkg, cb) {
+  var self = this;
```
Comment (jlipps) [inline lib/adb.js]: instead of aliasing `this` to `self`, we simply `.bind(this)` any function which needs the `this` context

**Comment by jlipps** [comment]
made another comment, otherwise looks good. and we'll ask you to squash everything into one commit, please

**Comment by hyeung-trulia** [comment]
I think messed the pull request up when trying to squash the commits. I created a new one with just the necessary changes: https://github.com/appium/appium-adb/pull/73

**Pull request closed without merging**